src/music_gui.cpp

```diff
@@ -834,6 +828,35 @@ struct MusicWindow : public Window {
 				break;
 		}
 	}
+
+	virtual void OnMouseDrag(Point pt, int widget)
+	{
+		this->UpdateVolumeWidget(pt);
+	}
+
+	void UpdateVolumeWidget(Point pt)
+	{
+		uint widget_width = this->GetWidget<NWidgetBase>(this->dragged_vol_widget)->current_x;
+
+		int x = pt.x - this->GetWidget<NWidgetBase>(this->dragged_vol_widget)->pos_x;
+
+		/* Only update the volume along the width of this widget */
+		if (x < 0) x = 0;
+		if (x > (int)widget_width) x = (int)widget_width;
+
+		byte *vol = (this->dragged_vol_widget == WID_M_MUSIC_VOL) ? &_settings_client.music.music_vol : &_settings_client.music.effect_vol;
+
+		byte new_vol = x * 127 / widget_width;
+		if (_current_text_dir == TD_RTL) new_vol = 127 - new_vol;
+		/* Clamp to make sure min and max are properly settable */
+		if (new_vol > 124) new_vol = 127;
+		if (new_vol < 3) new_vol = 0;
+		if (new_vol != *vol) {
+			*vol = new_vol;
+			if (this->dragged_vol_widget == WID_M_MUSIC_VOL) MusicDriver::GetInstance()->SetVolume(new_vol);
```

Comment:
Looks like you set the music volume but never the sound effect volume?

Reply:
It does appear that way, but this part of the code is unchanged from how it's working on master: it looks like the effect volume is updated as a result of updating `*vol` to point to `new_vol`.

Updating this slider does in fact make update the effect volume .. but if you think this code should be made clearer we can go that route.

Reply:
Oh! Interesting. Maybe it would be useful to add a comment.

Reply:
Alright, i've added some comments to clarify this